src/phoenix/server/api/routers/v1/datasets.py

```diff
@@ -450,3 +516,100 @@ async def _parse_form_data(
         metadata_keys,
         file,
     )
+
+
+async def _get_dataset_download_csv(request: Request) -> Response:
+    encoded_dataset_id = request.path_params.get("datasetId")
+    if not encoded_dataset_id:
+        return Response(
+            content="Missing Dataset ID",
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+        )
+    dataset_global_id = GlobalID.from_id(encoded_dataset_id)
+    if dataset_global_id.type_name != Dataset.__name__:
+        return Response(
+            content=f"Invalid Dataset ID: {encoded_dataset_id}",
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+        )
+    encoded_dataset_version_id = request.path_params.get("datasetVersionId")
+    dataset_version_global_id = (
+        GlobalID.from_id(encoded_dataset_version_id) if encoded_dataset_version_id else None
+    )
+    if dataset_version_global_id and dataset_version_global_id.type_name != DatasetVersion.__name__:
+        return Response(
+            content=f"Invalid Dataset Version ID: {encoded_dataset_version_id}",
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+        )
+    dataset_id = int(dataset_global_id.node_id)
```

Comment:
Suggestion: The `from_global_id` utility might be useful here.

Reply:
It still has the check for None. So going to punt on this one until later